### PR TITLE
rsgraber/20231031 fix merge producer configs

### PR DESCRIPTION
**Pull request by rgraber**
fix: handle missing keys when merging producer configs

**Description:** Fixes a KeyError when trying to merge producer configurations where either 'enabled' or 'event_key_field' is not being overridden

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,6 +14,12 @@ Change Log
 Unreleased
 ----------
 
+[9.0.1] - 2023-10-32
+--------------------
+Changed
+~~~~~~~
+* Fixed key error in merging event producer configs
```
Comment (timmc-edx) [inline CHANGELOG.rst:21]:
```suggestion
* Fixed key error in merging event producer configs. Previously, setting only one of `enabled` or `event_key_field` would result in the other being overwritten with None.
```

Comment (rgraber) [reply CHANGELOG.rst:21]: Actually it would result in a KeyError. Will update.

Comment (timmc-edx) [reply CHANGELOG.rst]: Ah, right.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -14,6 +14,12 @@ Change Log
 Unreleased
 ----------
 
+[9.0.1] - 2023-10-32
```
Comment (timmc-edx) [inline CHANGELOG.rst:17]:
```suggestion
[9.0.1] - 2023-10-31
```

**Review by timmc-edx — APPROVED** [review]

**Pull request merged**